### PR TITLE
adds legacy disruption to invariant tests

### PR DESCRIPTION
**Pull request by deads2k**

### CODE REVIEW

**Review by dgoodwin — COMMENTED** [review]
Started reviewing the wrong one, but here's what I had so far for partial comments.

File: pkg/cmd/monitor_command/run_monitor_command.go [hunk 1]
```diff
@@ -131,60 +132,15 @@ func (opt *RunMonitorOptions) Run() error {
...
-
-		if err = sampler.TearDownInClusterMonitors(restConfig); err != nil {
-			fmt.Printf("Failed to write events from in-cluster monitors, err: %v\n", err)
-		}
-
-		err = monitor.UploadIntervalsToLoki(intervals)
-		if err != nil {
-			// Best effort, we do not want to error out here:
-			logrus.WithError(err).Warn("unable to upload intervals to loki")
-		}
```
[the remark this answers is not in the document]
Comment (dgoodwin) [reply pkg/cmd/monitor_command/run_monitor_command.go:161]: I didn't find it yet but it did look like loki was getting intervals for your rehersals so I probably just didn't find it yet.

[the remark this answers is not in the document]
Comment (deads2k) [reply pkg/cmd/monitor_command/run_monitor_command.go:161]: Currently in monitor.go.  Will adjust eventually.

File: pkg/defaultinvariants/types.go [hunk 2]
```diff
@@ -1,11 +1,85 @@
...
+	"github.com/openshift/origin/pkg/invariants/alert_serializer"
+	"github.com/openshift/origin/pkg/invariants/availability_image_registry"
+	"github.com/openshift/origin/pkg/invariants/availability_legacy_apiservers"
+	"github.com/openshift/origin/pkg/invariants/availability_service_loadbalancer"
+	"github.com/openshift/origin/pkg/invariants/clusterinfo_serializer"
+	"github.com/openshift/origin/pkg/invariants/disruption_serializer"
+	"github.com/openshift/origin/pkg/invariants/interval_serializer"
+	"github.com/openshift/origin/pkg/invariants/timeline_serializer"
+	"github.com/openshift/origin/pkg/invariants/trackedresources_serializer"
+	watchpods "github.com/openshift/origin/pkg/invariants/watch_pods"
```
[the remark this answers is not in the document]
Comment (deads2k) [reply pkg/defaultinvariants/types.go:17]: addressed, but the golang standard is still bad.

File: pkg/defaultinvariants/types.go [hunk 3]
```diff
@@ -1,11 +1,85 @@
...
+	invariantTests.AddRegistryOrDie(newUniversalInvariants())
+
+	invariantTests.AddInvariantOrDie("service-type-load-balancer-availability", "NetworkEdge", availability_service_loadbalancer.NewAvailabilityInvariant())
+
+	invariantTests.AddInvariantOrDie("image-registry-availability", "Image Registry", availability_image_registry.NewAvailabilityInvariant())
+
+	invariantTests.AddInvariantOrDie("apiserver-availability", "kube-apiserver", availability_legacy_apiservers.NewAvailabilityInvariant())
+
+	return invariantTests
+}
```
[the remark this answers is not in the document]
Comment (deads2k) [reply pkg/defaultinvariants/types.go:55]: > How will everything in pkg/synthetictests/event_junits.go ultimately relate to this code. 

I think they need to get ported to the InvariantTest interface.

File: pkg/invariants/additional_events_collector/invariant.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+			// so for now we'll skip everything else for summaries
+			//
+			// TODO: TRT will need to double check this longterm what they want
+			// to do with these extra files
+			if !strings.HasPrefix(d.Name(), "AdditionalEvents__") {
+				return nil
+			}
+			saved, _ := monitorserialization.EventsFromFile(path)
+			intervalsFromStorage = append(intervalsFromStorage, saved...)
+			return nil
```
[the remark this answers is not in the document]
Comment (deads2k) [reply pkg/invariants/additional_events_collector/invariant.go:58]: > I thought this got nuked per the readme file, what scenarios are we still writing additional events?

Don't know yet, need to check CI.  Twisty passages.

File: pkg/invariants/availability_legacy_library/invariant.go [hunk 5]
```diff
@@ -0,0 +1,163 @@
+package availability_legacy_library
```
[the remark this answers is not in the document]
Comment (deads2k) [reply pkg/invariants/availability_legacy_library/invariant.go:1]: addressed

File: pkg/defaultinvariants/types.go [hunk 6]
```diff
@@ -1,11 +1,85 @@
...
+		panic(fmt.Sprintf("unknown cluster stability level: %q", clusterStabilityDuringTest))
+	}
+}
+
+func newDefaultInvariants() invariants.InvariantRegistry {
+	invariantTests := invariants.NewInvariantRegistry()
+
+	invariantTests.AddRegistryOrDie(newUniversalInvariants())
+
+	invariantTests.AddInvariantOrDie("service-type-load-balancer-availability", "NetworkEdge", availability_service_loadbalancer.NewAvailabilityInvariant())
```
[the remark this answers is not in the document]
Comment (deads2k) [reply pkg/defaultinvariants/types.go:48]: updated

**Comment by deads2k** [comment]
`[sig-arch] Only known images used by tests` is the last remaining.  Not sure if it was me, but if so, it's all that's left.  yay

File: pkg/test/ginkgo/cmd_runsuite.go [hunk 7]
```diff
@@ -91,6 +94,15 @@ func (o *GinkgoRunSuiteOptions) BindTestOptions(flags *pflag.FlagSet) {
 	flags.IntVar(&o.Parallelism, "max-parallel-tests", o.Parallelism, "Maximum number of tests running in parallel. 0 defaults to test suite recommended value, which is different in each suite.")
 }
 
+func (o *GinkgoRunSuiteOptions) Validate() error {
+	switch o.ClusterStabilityDuringTest {
+	case string(Stable), string(Disruptive):
+	default:
+		return fmt.Errorf("unknown --cluster-stability, %q, expected Stable or Disruptive")
```
Comment (stbenjam) [inline pkg/test/ginkgo/cmd_runsuite.go:101]: Missing format parameter for %q here

File: pkg/monitor/monitor.go [hunk 8]
```diff
@@ -1,227 +1,296 @@
...
-	newAnnotations := newMetadata.GetAnnotations()
-	if newAnnotations == nil {
-		newAnnotations = map[string]string{}
+func (m *Monitor) SerializeResults(ctx context.Context, junitSuiteName, timeSuffix string) error {
+	fmt.Fprintf(os.Stderr, "Serializing results.\n")
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	// don't bound the intervals that we return
+	finalIntervals := m.recorder.Intervals(time.Time{}, time.Time{})
```
Comment (DennisPeriquet) [inline pkg/monitor/monitor.go:176]: what was this done?

**Pull request merged**